src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -239,10 +241,34 @@ public static long countBulkFileLoadModelsFromPath(
             cloudEncapsulationId,
             fileRefColumns,
             errors)) {
+      nodesStream
```

Comment:
I think this is what's causing errors in your tests.  Streams can only be read once so when you then do a `count()` below, it'll fail.  You may need to explicitly count in the `forEach` block

Reply:
You could do it in one stream, by using `peek()`, something like:
```
nodesStream.takeWhile(...)
   .peek(loadFileModel -> ...)
   .count();
```

Reply:
You could also not use streaming for all the steps here and write:
```
    List<BulkLoadFileModel> models = nodesStream.takeWhile(...).collect();
    for (BulkLoadFileModel model : models ) { ... }
    return models.count();
```